Add generate-versions script and workflow

nielsvanvelzen opened the pull request:
This was requested so we can use the versions of various repositories in the docs.

The workflow is currently untested, data is not used (yet).

Once merged a webhook will be configured to trigger the workflow when a release is tagged. This will then regenerate the versions file and commit it which in turn deploys it to the website.

nielsvanvelzen committed:
Add generate-versions script and workflow

BotBlake commented:
Thank you so much <3

nielsvanvelzen commented:
I just reviewed it myself but can't fix it until later, feel free to fix the first two. The Node.Js update should be done separately (after this one is merged).

BotBlake commented:
> until later

We will see who's "later" comes first 😆 
I would also look at it "later" 🗿

BotBlake committed:
Merge branch 'master' into nvv-release-file

BotBlake reviewed (approved):
I donno 🗿 
Looks good tho

andrewrabert committed:
Mitigate pull_request_target privilege escalation

Hotfix — replaces pull_request_target with pull_request to stop
granting write permissions and secrets to fork PRs. Some workflows
will break; can be fixed properly later.
